lib/SILGen/SILGenThunk.cpp

```diff
@@ -404,7 +404,7 @@ SILFunction *SILGenModule::getOrCreateAutoDiffClassMethodThunk(
   auto originalFnDeclRef = derivativeFnDeclRef.asAutoDiffOriginalFunction();
   // TODO(TF-685): Use principled thunk mangling.
   // Do not simply reuse reabstraction thunk mangling.
-  auto name = derivativeFnDeclRef.mangle() + "_vtable_entry_thunk";
+  auto name = "AD__" + derivativeFnDeclRef.mangle() + "_vtable_entry_thunk";
```

Comment:
AD vtable entry thunks were never mangled originally. Now that derivative functions are mangled, I added a prefix so that this won't be interpreted as a mangled name. SR-13508 tracks mangling AD thunks, which I will tackle next.